tools/build_defs/framework.bzl

```diff
@@ -231,9 +231,9 @@ def cc_external_rule_impl(ctx, attrs):
     define_variables = [
         set_cc_envs,
         "export EXT_BUILD_ROOT=##pwd##",
-        "export BUILD_TMPDIR=##tmpdir##",
-        "export EXT_BUILD_DEPS=##tmpdir##",
         "export INSTALLDIR=$$EXT_BUILD_ROOT$$/" + empty.file.dirname + "/" + lib_name,
+        "export BUILD_TMPDIR=$$INSTALLDIR$$/_build_tmpdir",
+        "export EXT_BUILD_DEPS=$$INSTALLDIR$$/_ext_build_deps",
```

Comment:
Actually I think it would be nice if this was outside of `$$INSTALLDIR$$`, within `$$EXT_BUILD_ROOT$$` (i.e. `##pwd##`). That way we don't need to care about clashing with source files. Additionally, we could still use a tempdir under this prefix.